src/textual/screen.py

```diff
@@ -351,9 +351,21 @@ def _binding_chain(self) -> list[tuple[DOMNode, BindingsMap]]:
                 check_consume_key = filter_namespace.check_consume_key
                 for key in list(bindings_map.key_to_bindings):
                     if check_consume_key(key, key_to_character(key)):
+                        # If the widget consumes the key (e.g. like an Input widget),
+                        # then remove the key from the bindings map.
                         del bindings_map.key_to_bindings[key]
+
             filter_namespaces.append(namespace)
 
+        keymap = self.app.get_keymap()
+        for namespace, bindings_map in namespace_bindings:
+            if keymap:
+                result = bindings_map.apply_keymap(keymap)
+                if result.clashed_bindings:
+                    self.app.post_message(
```

Comment:
This could result in sending potentially messages on every keypress. TBH I don't think that is worth it.

Reply:
Do you have any suggestions for an alternative? As a developer I really want to be able to handle a user clobbering bindings cleanly - at least informing them of it.

Reply:
Could you check for clashes outside of a keypress somehow?

Otherwise, how about calling a method on App and letting the dev decide how to proceed?

Reply:
Anything is possible 😄 but I suspect checking outside of a keypress would be quite a bit of work, and not sure if it's worth it at this point.

Calling a method on App seems like a nice approach.